src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,34 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattleWeaponTable
+}
+local initialState: Data = {}
+local BrickbattlePlayerStateSlice = Reflex.createProducer(initialState, {
+    setStatState = function(state, playerName: string, weapon: string, weaponCount: number, Amount: number)
```

Comment:
unsure if Amount: number is correct given we can have a variety of types (vector3, cframe, number, boolean, model)